vignettes/advs.Rmd

```diff
@@ -0,0 +1,313 @@
+
+adsl <- adsl %>%
+  mutate(
+    TRT01P = ARM,
+    TRT01A = ACTARM
+  )
+
+adsl_vars <- exprs(TRTSDT, TRTEDT, TRT01P, TRT01A)
+
+advs <- derive_vars_merged(
+  vs,
+  dataset_add = adsl,
+  new_vars = adsl_vars,
+  by_vars = exprs(STUDYID, USUBJID)
+)
+
+advs <- derive_vars_dt(advs, new_vars_prefix = "A", dtc = VSDTC)
+
+advs <-
+  derive_vars_dy(advs, reference_date = TRTSDT, source_vars = exprs(ADT))
+
+param_lookup <- tribble(
+  ~VSTESTCD, ~PARAMCD, ~PARAM, ~PARAMN,
+  "HEIGHT", "HEIGHT", "Height (cm)", 2,
+  "HIPCIR", "HIPCIR", "Hip Circumference (cm)", 3,
+  "WEIGHT", "WEIGHT", "Weight (kg)", 4,
+  "WSTCIR", "WSTCIR", "Waist Circumference (cm)", 5
+)
+attr(param_lookup$VSTESTCD, "label") <- "Vital Signs Test Short Name"
+
+advs <- derive_vars_merged_lookup(
+  advs,
+  dataset_add = param_lookup,
+  new_vars = exprs(PARAMCD),
+  by_vars = exprs(VSTESTCD)
+)
+
+advs <- mutate(
+  advs,
+  AVAL = VSSTRESN
+)
+```
+
+## Derive Additional Parameters (e.g. `BMI` for `ADVS`) {#derive_param}
+
+In metabolic trials, `BMI` is often calculated at source using collected height and weight for screening purposes. But while creating the `ADVS` dataset, we re-derive `BMI` from the collected height and weight values. This is done to ensure that the `BMI` is calculated consistently across all subjects and visits. 
+
+In this step, we create parameter `Body Mass Index` (BMI) for `ADVS` domain using the `derive_param_bmi()` function.
+Note that only variables specified in the `by_vars` argument will be populated in the newly 
+created records. Also note that if height is collected only once use `constant_by_vars` to specify the subject-level variable to merge on.  Otherwise BMI is only calculated for visits where both are collected.
+
+
+```{r eval=TRUE}
+# Removing BMI collected at source from the dataset
+advs <- advs %>% filter(!VSTESTCD == "BMI")
+
+advs <- derive_param_bmi(
+  advs,
+  by_vars = exprs(STUDYID, USUBJID, !!!adsl_vars, VISIT, VISITNUM, ADT, ADY, VSTPT, VSTPTNUM),
+  set_values_to = exprs(
+    PARAMCD = "BMI",
+    PARAM = "Body Mass Index (kg/m^2)",
+    PARAMN = 1
+  ),
+  get_unit_expr = VSSTRESU,
+  filter = VSSTAT != "NOT DONE" | is.na(VSSTAT),
+  constant_by_vars = exprs(USUBJID)
+)
+```
+
+```{r, eval=TRUE, echo=FALSE}
+dataset_vignette(
+  arrange(advs, USUBJID, VISITNUM, VSTPTNUM, ADT, PARAMCD),
+  display_vars = exprs(USUBJID, VSTESTCD, PARAMCD, PARAM, VISIT, AVAL),
+  filter = PARAMCD %in% c("BMI")
+)
+```
+
+```{r eval=TRUE, include=FALSE}
+advs <- restrict_derivation(
+  advs,
+  derivation = derive_var_extreme_flag,
+  args = params(
+    by_vars = exprs(STUDYID, USUBJID, PARAMCD),
+    order = exprs(ADT, VSTPTNUM, VISITNUM),
+    new_var = ABLFL,
+    mode = "last"
+  ),
+  filter = (!is.na(AVAL) & ADT <= TRTSDT)
+)
+
+advs <- derive_var_base(
+  advs,
+  by_vars = exprs(STUDYID, USUBJID, PARAMCD),
+  source_var = AVAL,
+  new_var = BASE
+)
+
+advs <- derive_var_chg(advs)
+
+advs <- derive_var_pchg(advs)
+```
+
```

Comment:
I recommend adding a section here explaining what the common endpoints in an obesity trail are, and how we recommend coding them into ADVS. This will help justify the `AVALCAT` and `CRITFL` sections that come after.